### PR TITLE
Fixes #108 : unshorten shortened links

### LINKED ISSUE #108: unshorten shortened links to link stubs
Texts in loklak search results may contain shortened links which point to loklak for unshortening, like: http://loklak.org/x?id=802029515690180608
We want to see those links as original link stubs. To make this possible, the loklak json now contains an ```unshorten``` object, i.e.
```
"text": "Please RT #interiordesign #decoration #architecture #home Best Friday Five Posts of 2015 http://loklak.org/x?id=802029515690180608 https://pic.twitter.com/9EVncMj6AG",
"unshorten": {"http://loklak.org/x?id=802029515690180608": "http://decorationandstyle.com/decoration/..."}
```
Here, the link http://loklak.org/x?id=802029515690180608 is currently rendered as
```
<a href="http://loklak.org/x?id=802029515690180608">http://loklak.org/x?id=802029515690180608</a>
```
but using the ```unshorten``` object, it is possible to translate the link back into ```http://decorationandstyle.com/decoration/...```
The resulting link in the html should be
```
<a href="http://loklak.org/x?id=802029515690180608">http://decorationandstyle.com/decoration/...</a>
```

To do:
- for each key in the ```unshorten``` object, try to match this with a substring in ```text````
- replace the link text for that link with the value of the object in ```unshorten```

### PR DESCRIPTION
**Pull request by shubhsingh594**
1)adds a new variable "unshorten" to resolve the parsing of the JSON object

2)Replaces the shortened link with the original one

Fixes #108

### CODE REVIEW

File: src/app/shared/classes/search/api-response.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ export class ApiResponseResult {
 		public screen_name: string = null,
 		public source_type: string = null,
 		public text: string = null,
+		public unshorten:any =null,
```
Comment (hemantjadon) [inline src/app/shared/classes/search/api-response.ts]: Can It be more specific type than `any` like `JSON` or `Object` ?

**Comment by hemantjadon** [comment]
Please also check for the linting errors, you can run `ng lint` locally on terminal for quick check without explicitly commiting.

File: src/app/shared/classes/search/api-response.ts (hunk 1 above)
Comment (shubhsingh594) [inline src/app/shared/classes/search/api-response.ts]: yeah sure! will make the changes

**Comment by hemantjadon** [comment]
Please also check the Review along with the two remaining linting errors.

**Comment by hemantjadon** [comment]
@shubhsingh594 
This seems nice to me.
It will be great if you can squash the commits into one.
https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/ Might be helpful.

Cheers :)

**Comment by hemantjadon** [comment]
This seems pretty good from my side.
@daminisatya Can you please look into this and proceed further.

**Comment by Orbiter** [comment]
please do not merge: from what I see in the code change there is just a simple replacement of the shortlink with the unshortened link. This does not work, because the 'unshorten' object is a mapping to link stubs: these Strings are not actually URLs.

If you read the requirement, these links shall only replace the anchor text, not the anchor link. This commit will change both. That will cause broken links.

**Comment by hemantjadon** [comment]
@Orbiter  Ohh You are absolutely correct, this just slipped of my mind.

**Pull request closed without merging**